src/ArticleController.m

```diff
@@ -615,11 +615,42 @@ -(void)markDeletedByArray:(NSArray *)articleArray deleteFlag:(BOOL)deleteFlag
 	NSMutableArray * folderArrayCopy = [NSMutableArray arrayWithArray:folderArrayOfArticles];
 	__block BOOL needReload = NO;
 	
+	NSString * guidToSelect = nil;
+	
     // if we mark deleted, mark also read and unflagged
 	if (deleteFlag) {
 	    [self innerMarkReadByRefsArray:articleArray readFlag:YES];
         [self innerMarkFlaggedByArray:articleArray flagged:NO];
-		[mainArticleView selectPreviousArticle];
+		
+		Article * firstArticle = articleArray.firstObject;
+		if (firstArticle != nil) // Should always be true
+		{
+			// We want to select the next non-deleted article
+			NSUInteger articleIndex = [currentArrayOfArticles indexOfObject:firstArticle];
+			if (articleIndex != NSNotFound)
+			{
+				NSUInteger count = currentArrayOfArticles.count;
+				for (NSUInteger i = articleIndex + 1; i < count; ++i)
+				{
+					Article * nextArticle = [currentArrayOfArticles objectAtIndex:i];
+					if (![articleArray containsObject:nextArticle])
+					{
+						guidToSelect = nextArticle.guid;
+						break;
+					}
+				}
+				
+				// Otherwise, we want to select the previous article.
+				if (guidToSelect == nil && articleIndex > 0)
+				{
+					Article * nextArticle = [currentArrayOfArticles objectAtIndex:articleIndex - 1];
+					guidToSelect = nextArticle.guid;
+				}
+				
+				// Deselect all now, select article after refresh
```

Comment:
This comment is ambiguous : I’d prefer ”Deselect all now, will select article after refresh“.
I am going to approve the PR, but will fix the comment when porting to master